coderd/batchstats/batcher.go

```diff
@@ -156,9 +159,13 @@ func (b *Batcher) Add(
 	b.buf.SessionCountSSH = append(b.buf.SessionCountSSH, st.SessionCountSSH)
 	b.buf.ConnectionMedianLatencyMS = append(b.buf.ConnectionMedianLatencyMS, st.ConnectionMedianLatencyMS)
 
-	// If the buffer is full, signal the flusher to flush immediately.
-	if len(b.buf.ID) == cap(b.buf.ID) {
+	// If the buffer is over 80% full, signal the flusher to flush immediately.
+	// We want to trigger flushes early to reduce the likelihood of
+	// accidentally growing the buffer over batchSize.
+	filled := float64(len(b.buf.ID)) / float64(b.batchSize)
+	if filled >= 0.8 && !b.flushForced.Load() {
 		b.flushLever <- struct{}{}
+		b.flushForced.Store(true)
```

Comment:
Does the `flushForced` just prevent a deadlock by adding to the `flushLever` channel if it already has something?

Can we just use a select statement to prevent pushing to the flushLever channel if it already has an item?

Reply:
I tried the select but it will still end up doing one extra unwanted flush.

Reply:
Well, in theory an extra flush is lesser evil than losing data.